src/components/@shared/FormInput/InputElement/ConsumerParameters/_validation.ts

```diff
@@ -0,0 +1,40 @@
+import { FormConsumerParameter } from '@components/Publish/_types'
+import * as Yup from 'yup'
+import { SchemaLike } from 'yup/lib/types'
+import { paramTypes } from '.'
+
+export const validationConsumerParameters: {
+  [key in keyof FormConsumerParameter]: SchemaLike
+} = {
+  name: Yup.string()
+    .test('unique', 'Parameter names must be unique', (name, context) => {
+      // TODO: revert any
+      // from is not yet correctly typed: https://github.com/jquense/yup/issues/398#issuecomment-916693907
```

Comment:
Can you resolve this TODO?

Reply:
The `from` property is correctly typed in Yup v1.0  but not in v0.32.11 the market uses. We could manually extend the TestContext type with something like:

```
interface TestContextExtended extends TestContext {
  from: {
    value: FormPublishData['metadata'] | FormPublishService
  }[]
}
```

and then 

```
const [parentFormObj, nextParentFormObj] = (
        context as TestContextExtended
      ).from
```

Would that work for you?

Reply:
Fair enough, yeah that seems like a good solution